source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -431,25 +443,25 @@ void Wavm::start(Context* context) {
   }
 }
 
-absl::string_view Wavm::getMemory(uint32_t pointer, uint32_t size) {
+absl::string_view Wavm::getMemory(uint64_t pointer, uint64_t size) {
```

Comment:
I am hitting a really weird memory related bug when compile the code to wasm module and I think it is related to this getMemory/setMemory call. Could you help me understand how does these 64 bit pointers get handled by WAVM? What does the `WasmUntaggedValue` do? Does it affect how WAVM deal with these ptrs? Thanks!

cc @mandarjog

Reply:
The pointer here is just the bit image of a normal pointer in envoy for the Null sandbox and it is the offset into Memory for WAVM or V8.   We manipulate them as opaque uint64_t in the API implementation.  Now some of these pointers are pointers to pointers.  In this case we use setWord so that the correct size pointer (64-bits for Null, 32-bits for WASM) are set.

Reply:
Please send me instructions on how to repro, thanx!

Reply:
fixed